python/hail/ir/ir.py

```diff
@@ -403,6 +715,14 @@ def __init__(self, a, constructor_args, init_op_args, agg_sig):
         self.init_op_args = init_op_args
         self.agg_sig = agg_sig
 
+    def copy(self, *args):
```

Comment:
I'm curious, why not use a `copy(a, constructor_args, init_op_args)` signature here?

Reply:
Children is passed as a single list. The caller doesn't know which args correspond to constructor args or init op args.

Reply:
Oh I get it, thanks.